dev/benchmarks/complex_layout/lib/main.dart

```diff
@@ -36,13 +37,45 @@ class ComplexLayoutAppState extends State<ComplexLayoutApp> {
     });
   }
 
+  ScrollMode _scrollMode = ScrollMode.complex;
+  ScrollMode get scrollMode => _scrollMode;
+  set scrollMode(ScrollMode mode) {
+    setState(() {
+      _scrollMode = mode;
+    });
+  }
+
   void toggleAnimationSpeed() {
     setState(() {
       timeDilation = (timeDilation != 1.0) ? 1.0 : 5.0;
     });
   }
 }
 
+class TileScrollLayout extends StatelessWidget {
+  const TileScrollLayout({ Key key }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[];
+    for (int i = 0; i < 200; i++) {
```

Comment:
For 200 objects, it most likely is a common case. If we are talking about an unlimited list (such as an email client), then it would be worth implementing a lazy data loading model layer. That kind of functionality is complex enough that most app devs won't bother with it if the number of items is somewhat bounded.

So, given that the entire list of model objects is loaded up front, the question is whether to build the ListView contents eagerly or using a delegate. This is where it gets interesting.

On Android, ListView does not do viewporting. RecyclerView does. If you use RecyclerView, you *have to* use a delegate to lazy load views (https://developer.android.com/reference/android/support/v7/widget/RecyclerView.Adapter.html). The API does not support any other way. I understood why it was done this way but it did create complexity in our app code.

On Flutter, ListView does viewporting and you can build its children eagerly. That leads to much simpler app code and viewporting is done behind the scenes. Therefore, I believe given the options that exist in Flutter, that's what most devs will choose to do since it leads to most straight-forward and readable code.